src/sbvr-api/permissions.ts

```diff
@@ -430,17 +481,112 @@ const generateConstrainedAbstractSql = (
 	const collapsedPermissionFilters = collapsePermissionFilters(
 		permissionFilters,
 	);
-	_.set(odata, ['tree', 'options', '$filter'], collapsedPermissionFilters);
 
-	const translated = uriParser.translateUri({
-		method: 'GET',
-		resourceName,
+	return collapsedPermissionFilters;
+};
+
+const constrainedPermissionError = new PermissionError();
+const generateConstrainedAbstractSql = (
+	permissionsLookup: PermissionLookup,
+	actionList: PermissionCheck,
+	vocabulary: string,
+	resourceName: string,
+) => {
+	const abstractSQLModel = sbvrUtils.getAbstractSqlModel({
 		vocabulary,
-		odataBinds: odata.binds,
-		odataQuery: odata.tree,
-		values: {},
 	});
-	const abstractSqlQuery = _.clone(translated.abstractSqlQuery!);
+	const odata = memoizedParseOdata(`/${resourceName}`);
+
+	const collapsedPermissionFilters = buildODataPermission(
+		permissionsLookup,
+		actionList,
+		vocabulary,
+		resourceName,
+		odata,
+	);
+
+	_.set(odata, ['tree', 'options', '$filter'], collapsedPermissionFilters);
+
+	const lambdaAlias = randomstring.generate(20);
+	let inc = 0;
+
+	// We need to trace the processed resources, to be able to break
+	// permissions circles.
+	const canAccessTrace: string[] = [resourceName];
+
+	const canAccessFunction: ResourceFunction = function(property: AnyObject) {
+		// remove method property so that we won't loop back here again at this point
+		delete property.method;
+
+		if (!this.defaultResource) {
+			throw new Error(`No resource selected in AST.`);
+		}
+
+		const targetResource = this.NavigateResources(
+			this.defaultResource,
+			property.name,
+		);
+
+		const targetResourceName = sqlNameToODataName(targetResource.resource.name);
+
+		if (canAccessTrace.includes(targetResourceName)) {
+			// return a boolen subquery
```

Comment:
```suggestion
			// return a boolean subquery that is always false
```
(or `true` and update the returned query if that was the real intention)